From: Fabian Vogt <fvogt@suse.de>
Date: Mon, 21 Aug 2023 14:17:01 +0200
Subject: [PATCH] dracut: Don't include the ignition module by default

Currently the module is automatically included in all initrds, hostonly or
generic. Leave it to the distro provided module to pull it in explicitly.
---
diff --git a/dracut/30ignition/module-setup.sh b/dracut/30ignition/module-setup.sh
index ad7e80fd..f431b7dc 100755
--- a/dracut/30ignition/module-setup.sh
+++ b/dracut/30ignition/module-setup.sh
@@ -2,6 +2,13 @@
 # -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
 # ex: ts=8 sw=4 sts=4 et filetype=sh
 
+check() {
+    # Only include this if another module requests it.
+    # In our case it'll be the distro provided module with integration and customizations
+    # (coreos-ignition/ignition-microos/...).
+    return 255
+}
+
 depends() {
     echo qemu systemd url-lib network
 }
-- 
2.41.0

